tests/recovery/recovery_test/recovery_test.go

```diff
@@ -0,0 +1,108 @@
+//go:build darwin || (linux && amd64)
+// +build darwin linux,amd64
+
+package recovery
+
+import (
+	"errors"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/kopia/kopia/tests/recovery/blobmanipulator"
+	"github.com/kopia/kopia/tests/robustness/snapmeta"
+	"github.com/kopia/kopia/tests/tools/kopiarunner"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSnapshotFix(t *testing.T) {
+	// assumption: the test is run on filesystem and not directly on object store
+	dataRepoPath := path.Join(*repoPathPrefix, dataSubPath)
+
+	baseDir := makeDir("base-dir-")
+
+	bm, err := blobmanipulator.NewBlobManipulator(baseDir)
+	if err != nil {
+		if errors.Is(err, kopiarunner.ErrExeVariableNotSet) {
+			log.Println("Skipping robustness tests because KOPIA_EXE is not set")
+		} else {
+			log.Println("Error creating Blob Manipulator:", err)
+		}
+	}
+
+	bm.DirCreater = getSnapshotter(baseDir)
+	bm.ConnectOrCreateRepo(dataRepoPath)
+	bm.DataRepoPath = dataRepoPath
+
+	// populate the kopia repo under test with random snapshots
+	bm.SetUpSystemUnderTest()
+
+	// delete random blob
+	// assumption: the repo contains "p" blobs to delete, else the test is a no-op
```

Comment:
Is it ever possible for this to happen? If so, will the rest of the test run/fail as expected?

Reply:
Handled this in the code